text/3729-sized-hierarchy.md

```diff
@@ -0,0 +1,1412 @@
+on `Sized` traits resolves blockers for other RFCs which have had significant interest.
+
+## Runtime-sized types and scalable vector types
+[runtime-sized-types-and-scalable-vector-types]: #runtime-sized-types-and-scalable-vector-types
+
+Rust already supports [SIMD][rfc_simd] (*Single Instruction Multiple Data*),
+which allows operating on multiple values in a single instruction. Processors
+have SIMD registers of a known, fixed length and a variety of intrinsics
+which operate on these registers. For example, x86-64 introduced 128-bit SIMD
+registers with SSE, 256-bit SIMD registers with AVX, and 512-bit SIMD registers
+with AVX-512, and Arm introduced 128-bit SIMD registers with Neon.
+
+As an alternative to releasing SIMD extensions with greater bit widths, Arm and
+RISC-V have vector extensions (SVE/Scalable Vector Extension and the "V" Vector
+Extension/RVV respectively) where the bit width of vector registers depends on the
+CPU implementation, and the instructions which operate these registers are bit
+width-agnostic.
+
+As a consequence, these types are not `Sized` in the Rust sense, as the size of
+a scalable vector cannot be known during compilation, but is a runtime constant.
+For example, the size of these types could be determined by inspecting the value
+in a register - this is not available at compilation time and the value may
+differ between any given CPU implementation. Both SVE and RVV have mechanisms to
+change the system's vector length (up to the maximum supported by the CPU
+implementations) but this is not supported by the proposed ABI for these types.
+
+However, despite not implementing `Sized`, these are value types which should
+implement `Copy` and can be returned from functions, can be variables on the
+stack, etc. These types should implement `Copy` but given that `Sized` is a
+supertrait of `Copy`, they cannot be `Copy` without being `Sized`, and
+they aren't `Sized`.
+
+Introducing a `const Sized` trait will enable `Copy` to be implemented for
+those types whose size is a runtime constant to function correctly without
+special cases in the type system. See
+[rfcs#3268: Scalable Vectors][rfc_scalable_vectors].
+
+## Unsized types and extern types
+[unsized-types-and-extern-types]: #unsized-types-and-extern-types
+
+[Extern types][rfc_extern_types] has long been blocked on these types being
+neither `Sized` nor `?Sized` ([relevant issue][issue_extern_types_align_size]).
+
+RFC #1861 defined that `std::mem::size_of_val` and `std::mem::align_of_val`
+should not be defined for extern types but not how this should be achieved, and
+suggested an initial implementation could panic when called with extern types,
+but this is always wrong, and not desirable behavior in keeping with Rust's
+values. `size_of_val` returns 0 and `align_of_val` returns 1 for extern types in
+the current implementation. Ideally `size_of_val` and `align_of_val` would error
```

Comment:
That's not correct, I made them panic a while ago: https://github.com/rust-lang/rust/pull/118534

Reply:
Fixed!